web/third/marked/lib/marked.d.ts

```diff
@@ -0,0 +1,42 @@
+/*
+This is an incomplete stub, which we include here because
+we only use parts of the marked API and are using a fork
+of the upstream library.
+*/
+
+import type {ParseOptions} from "../../../src/markdown";
+
+export class Renderer {
+    code: (code: string) => string;
+    link: (href: string, title: string, text: string) => void;
+    br: () => string;
+}
+
+export type RegExpOrStub = RegExp | {
+    exec: () => boolean;
+};
```

Comment:
This needs a `string` argument.

Also we should probably fix the return type to match the real `RegExp`, and adjust the stubs to match:

```suggestion
export type RegExpOrStub = {
    exec(string: string): RegExpExecArray | null;
};
```

Reply:
I think we'd also need to change this then

```js
    function disable_markdown_regex(rules: Record<string, RegExpOrStub>, name: string): void {
        rules[name] = {
            exec() {
                return false;
            },
        };
    }
```

should it be this instead?

```js
        rules[name] = {
            exec(_: string) {
                return null;
            },
        };
```

Once I switched that, things seemed to work okay.

I'll note also that when I tried `exec: RegExp["exec"]` I got this error:

```
Type '{ exec(_: string): null; }' is not assignable to type 'RegExpOrStub'.
  Type '{ exec(_: string): null; }' is not assignable to type '{ exec(string: string): (string: string) => RegExpExecArray | null; }'.
    The types returned by 'exec(...)' are incompatible between these types.
      Type 'null' is not assignable to type '(string: string) => RegExpExecArray | null'.
```